daemon/containerd/image_snapshot.go

```diff
@@ -67,3 +70,41 @@ func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, parentIma
 	_, err = s.Prepare(ctx, id, parent)
 	return err
 }
+
+// calculateSnapshotParentUsage returns the usage of all ancestors of the
+// provided snapshot. It doesn't include the size of the snapshot itself.
+func calculateSnapshotParentUsage(ctx context.Context, snapshotter snapshots.Snapshotter, snapshotID string) (snapshots.Usage, error) {
+	info, err := snapshotter.Stat(ctx, snapshotID)
+	if err != nil {
+		return snapshots.Usage{}, errors.Wrapf(err, "snapshotter.Stat failed for %s", snapshotID)
+	}
+	if info.Parent == "" {
+		return snapshots.Usage{}, errdefs.NotFound(errors.Wrapf(err, "snapshot %s has no parent", snapshotID))
+	}
+
+	return calculateSnapshotTotalUsage(ctx, snapshotter, info.Parent)
+}
+
+// calculateSnapshotTotalUsage returns the total usage of that snapshot
+// including all of its ancestors.
+func calculateSnapshotTotalUsage(ctx context.Context, snapshotter snapshots.Snapshotter, snapshotID string) (snapshots.Usage, error) {
+	var total snapshots.Usage
+	next := snapshotID
+
+	for next != "" {
+		usage, err := snapshotter.Usage(ctx, next)
+		if err != nil {
+			return total, errors.Wrapf(err, "snapshotter.Usage failed for %s", next)
```

Comment:
Same here (or is `snapshotter.Usage` already returning a typed error that we can use?